onnxruntime/core/providers/cuda/reduction/reduction_ops.cc

```diff
@@ -99,12 +101,18 @@ Status ReduceKernel<allow_multi_axes>::ComputeImpl(OpKernelContext* ctx, cudnnRe
   }
 
   if (keepdims_) {
+    // since keepdims is set, the final output dim is the same as the output_dims computed above
     squeezed_output_dims = output_dims;
-  } else {
+  } else if (axes_.size() > 0) {
+    // we are not going to keep the reduced dims, hence compute the final output dim accordingly
+    squeezed_output_dims.reserve(rank);  // even though we won't use the full capacity, it is better to reserve for peak possible usage
     for (size_t i = 0; i < rank; ++i) {
       if (!reduced[i])
         squeezed_output_dims.push_back(input_dims[i]);
     }
+  } else {
+    // 'axes' is empty and keepdims is false => we reduce on all axes AND drop all dims,
```

Comment:
we already validate to make sure the input is non-empty in line 94 because we can't have an empty input and reduce on all dims while dropping all the dims to produce a scalar. But if we have reached this line of execution, we have already validated to make sure the input is non-empty...